mono/metadata/marshal-ilgen.c

```diff
@@ -6220,6 +6220,26 @@ signature_param_uses_handles (MonoMethodSignature *sig, MonoMethodSignature *gen
 		return ICALL_HANDLES_WRAP_NONE;
 }
 
+static void
+mono_emit_handle_raw (MonoMethodBuilder *mb)
+{
+	// MONO_HANDLE_RAW()
+	//
+	// This is playing a bit loose with the types.
+	// For most of the C code, there is a struct passed by value,
+	// with a pointer to a pointer. For this code, in order to ease
+	// ABI concerns, it is a pointer to a struct containing a pointer.
+	//
+	// The same level of indirection either way.
+	//
+	// Generated C code converts between the types. The JIT doesn't
+	// have to pass/accept structs by value.
+	//
+	// In either case, we dereference twice and both offsets are 0.
+	mono_mb_emit_ldflda (mb, MONO_HANDLE_OFFSET (MonoObject));
```

Comment:
I don't understand this, we said we'd switch to use an opaque `void*` for marshaling handles yet we're using the code for structs.

Reply:
Yea the comment and the code are wrong. There are no structs here.

This could just needs to implement: `(MonoObject*)*((gpointer*)raw_handle))` - take the raw handle, dereference it, and cast the result to a `MonoObject*`.

Reply:
There is meant to be no change to representation at this time -- i.e. marshal the old way, a pointer to a struct containing a pointer. However, the struct is no longer used or declared anywhere else, so this code just leaves kind of a note and does things oddly, in case the struct later changes (which granted, does not make this code keep working).

We could either declare the old Payload struct and use it only here.
Or just treat this as a void** and deref once -- I think the first instruction is effectively just a nop -- a type-changing move.

I will try just removing the first instruction, and remove the rest.